sklearn/neighbors/base.py

```diff
@@ -79,8 +79,16 @@ def _get_weights(dist, weights):
     if weights in (None, 'uniform'):
         return None
     elif weights == 'distance':
-        with np.errstate(divide='ignore'):
-            dist = 1. / dist
+        # if user attempts to classify a point that was zero distance from one
+        # or more training points, those training points are weighted as 1.0
+        # and the other points as 0.0
+        for point_dist_i, point_dist in enumerate(dist):
```

Comment:
It turned out not to be expensive and try to special case the `KNeighbors*` made it worse.  See my latest comment in the PR discussion.